src/main/java/uk/gov/hmcts/reform/sscs/service/ccdupdate/HearingsJourneyService.java

```diff
@@ -0,0 +1,87 @@
+    private static final String CANCELLED_SUMMARY = "SSCS - case cancelled";
+    private static final String CANCELLED_DESC = "SSCS - case cancelled";
+    private static final String EXCEPTION_SUMMARY = "SSCS - exception occurred";
+    private static final String EXCEPTION_DESC = "SSCS - exception occurred";
+
+    private final HmcHearingService hmcHearingService;
+    private final CcdCaseService ccdCaseService;
+    private final CcdStateUpdateService ccdStateUpdateService;
+    private final CcdLocationUpdateService locationUpdateService;
+
+    public HearingsJourneyService(HmcHearingService hmcHearingService,
+                                  CcdCaseService ccdCaseService,
+                                  CcdStateUpdateService ccdStateUpdateService,
+                                  CcdLocationUpdateService locationUpdateService) {
+        this.hmcHearingService = hmcHearingService;
+        this.ccdCaseService = ccdCaseService;
+        this.ccdStateUpdateService = ccdStateUpdateService;
+        this.locationUpdateService = locationUpdateService;
+    }
+
+    public void process(HmcMessage hmcMessage) throws GetCaseException, UpdateCaseException {
+        validateHmcMessage(hmcMessage);
+
+        final String hearingId = hmcMessage.getHearingID();
+        HearingGetResponse hearingResponse = hmcHearingService.getHearingRequest(hearingId);
+        if (isNull(hearingResponse)) {
+            throw new GetCaseException(String.format("Failed to retrieve hearing with Id: %s from HMC", hearingId));
+        }
+
+        SscsCaseData caseData = ccdCaseService.getCaseDetails(Long.parseLong(hearingId)).getData();
+
+        // todo review statuses
+        switch (hmcMessage.getHearingUpdate().getHmcStatus()) {
+            case "LISTED":
+                ccdStateUpdateService.updateListed(hearingResponse, caseData);
+                locationUpdateService.updateVenue(hmcMessage, caseData);
+                ccdCaseService.updateCaseDetails(caseData, NEW_HEARING_BOOKED, CREATED_SUMMARY, CREATED_DESC);
+                break;
+            case "UPDATE_SUBMITTED":
+                ccdStateUpdateService.updateListed(hearingResponse, caseData);
+                locationUpdateService.updateVenue(hmcMessage, caseData);
+                ccdCaseService.updateCaseDetails(caseData, UPDATE_CASE_ONLY, UPDATED_SUMMARY, UPDATED_DESC);
+                break;
+            case "CANCELLED":
+                ccdStateUpdateService.updateCancelled(hearingResponse, caseData);
+                ccdCaseService.updateCaseDetails(caseData, UPDATE_CASE_ONLY, CANCELLED_SUMMARY, CANCELLED_DESC);
+                break;
+            case "EXCEPTION":
+                ccdStateUpdateService.updateFailed(caseData);
+                ccdCaseService.updateCaseDetails(caseData, UPDATE_CASE_ONLY, EXCEPTION_SUMMARY, EXCEPTION_DESC);
+                break;
+            default:
+                break;
```

Comment:
Should probably log or throw something here

Reply:
maybe log an info level? - in my opinion it's not actually an error if someone calls this method with the Hmc status witch 
 does not trigger a state update in CCD. But it's your call.